--- a/tools/setup_helpers/cmake.py
+++ b/tools/setup_helpers/cmake.py
@@ -273,6 +273,13 @@
                     "TORCH_XPU_ARCH_LIST",
                     "TRACING_BASED",
                     "PYTHON_LIB_REL_PATH",
+                    "Python_INCLUDE_DIR",
+                    "Python_LIBRARY",
+                    "Python_NumPy_INCLUDE_DIR",
+                    "Python_NumPy_INCLUDE_DIRS",
+                    "Python_ROOT_DIR",
+                    "Python_FIND_VIRTUALENV",
+                    "Python_FIND_STRATEGY",
                 )
             }
         )
