diff --git a/functorch/CMakeLists.txt b/functorch/CMakeLists.txt
--- a/functorch/CMakeLists.txt
+++ b/functorch/CMakeLists.txt
@@ -9,6 +9,13 @@
 file(GLOB_RECURSE FT_SOURCES ${FT_DIR}/*.cpp ${FT_DIR}/*.c)
 
 add_library(${PROJECT_NAME} MODULE ${FT_SOURCES})
+if(USE_ROCM)
+  foreach(_c_src ${FT_SOURCES})
+    if(_c_src MATCHES "\.c$")
+      set_source_files_properties(${_c_src} PROPERTIES LANGUAGE C COMPILE_FLAGS "-std=gnu11 -fno-strict-aliasing -Wno-strict-aliasing")
+    endif()
+  endforeach()
+endif()
 target_include_directories(${PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 target_compile_definitions(${PROJECT_NAME} PRIVATE FUNCTORCH_BUILD_MAIN_LIB)
 target_compile_definitions(${PROJECT_NAME} PRIVATE TORCH_EXTENSION_NAME=_C)
